src/Strategy/Node/YarnV1/YarnLock.hs

```diff
@@ -85,12 +85,14 @@ buildGraph ::
   FlatDeps ->
   m (Graphing Dependency)
 buildGraph lockfile FlatDeps{..} = fmap hydrateDepEnvs . withLabeling toDependency $
-  for_ (map firstKey $ MKM.toList lockfile) $ \(key, pkg) -> do
+  for_ (map allkeys $ MKM.toList lockfile) $ \(firstKey, keys, pkg) -> do
```

Comment:
[suggest]: If we use a better extraction function like the suggested `extractPkgTriple`, then this needs to change accordingly:
```suggestion
  for_ (map extractPkgTriple $ MKM.toList lockfile) $ \(pkgName, keys, pkg) -> do
```